Orange/widgets/data/owtranspose.py

```diff
@@ -57,20 +54,23 @@ def _feature_combo_changed(self):
         self.apply()
 
     def set_data(self, data):
-        self.closeContext()
+        # Skip the context if the combo is empty: a context with
```

Comment:
I spent an hour hunting this down. The problem is that if you take Iris, it sets `feature_names_column` to `None`. The you take zoo. The Iris' context matches for zoo domain since it doesn't require any attribute - hence it will set `feature_names_column` to `None`. For the bug to be harder to track, the combo will not reflect this since it does not contain a `None` value.

This is not a bug in `DomainContextHandler`, but it will appear in all such cases. @astaric, what about preventing a match when no attribute were matched since no matcher were required?

`PerfectDomainContextHandler` would work here. :) But I still think that widgets shouldn't use `PerfectDomainContextHandler` unless unavoidable.